lib/active_shipping/carriers/ups.rb

```diff
@@ -164,6 +169,16 @@ def create_shipment(origin, destination, packages, options = {})
       end
     end
 
+    def get_delivery_dates(origin, destination, packages, pickup_date=Date.current, options = {})
```

Comment:
Can you move pickup date into `options`? This way the method call is compatible with `find_rates`, which would allow us to merge them at some point. Given that it's optional it makes more sense to have in the options hash anyway.